hpx/lcos/detail/future_data.hpp

```diff
@@ -423,6 +428,7 @@ namespace detail
 
             // check whether the data has already been set
             if (is_ready_locked()) {
+                l.unlock();
```

Comment:
THROWS might suspend as the stack is collected on a different HPX thread under certain circumstances, this has nothing to do with the problem at hand, it's just a fly-by.